flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1487,7 +1487,7 @@ class JobManager(
           }
         }
 
-        eg.fail(cause)
+        eg.cancel()
```

Comment:
Yes, that would work during shutdown, but there will be a chance that a `fail` right before `cancelAndClearEverything` will still result in the restarting behaviour, because multiple calls to `fail` are ignored when the job status is `FAILING`. `cancel` makes sure that this does not happen, because cancellation "overwrites" failing behaviour.

If we say that this is OK as a corner case, we can keep the `fail` on `cancelAndClearEverything` and wrap the Exception to suppress restarts in the common case.